### PR TITLE
Do not loop over boundary nodes when there is no NodalBC

### LINKED ISSUE #28254: Do not loop over boundary nodes when there is no NodalBC
### Motivation

`NonlinearSystemBase::computeNodalBCs` loops over boundary nodes whenever there are boundary nodes (including reinit), regardless of whether there is any `NodalBC` in the system. This incurs some overhead and can be avoided easily.

### Design

`NonlinearSystemBase::computeNodalBCs` will return early when the size of `_nodal_bcs` is zero, and the loop will not be executed when `nbc_warehouse->size()` is zero.

### Impact

Reduced overhead.

### PR DESCRIPTION
**Pull request by NamjaeChoi**
Closes #28254.

### CODE REVIEW

File: framework/src/systems/NonlinearSystemBase.C [hunk 1]
```diff
@@ -2061,24 +2064,25 @@ NonlinearSystemBase::computeNodalBCs(const std::set<TagID> & tags)
       else
         nbc_warehouse = &(_nodal_bcs.getVectorTagsObjectWarehouse(tags, 0));
 
-      for (const auto & bnode : bnd_nodes)
-      {
-        BoundaryID boundary_id = bnode->_bnd_id;
-        Node * node = bnode->_node;
-
-        if (node->processor_id() == processor_id())
+      if (nbc_warehouse->size())
```
Comment (lindsayad) [inline framework/src/systems/NonlinearSystemBase.C:2067]: I think the early check should be enough. What I would do is move the creation of `nbc_warehouse` pointer ahead of your early return check, and so then do 
```c++
if (!nbc_warehouse->size())
  return;
```
And then you don't need the later `if` checks that you've added

Comment (lindsayad) [reply framework/src/systems/NonlinearSystemBase.C:2067]: It would reduce the diff too as a nice byproduct

**Review by lindsayad — CHANGES_REQUESTED** [review]

File: framework/src/systems/NonlinearSystemBase.C (hunk 1 above)
Comment (NamjaeChoi) [reply framework/src/systems/NonlinearSystemBase.C:2067]: Sounds good.

Comment (NamjaeChoi) [reply framework/src/systems/NonlinearSystemBase.C:2067]: Do you think I can also move `reinitNodeFace` inside the following `if`?

**Comment by NamjaeChoi** [comment]
The controlled app test failure is unrelated.

**Review by lindsayad — APPROVED** [review]

**Pull request merged**